internal/database/postgres.go

```
@@ -231,6 +231,15 @@ func (db *PostgreSQL) UpdateServer(ctx context.Context, id string, server *apiv0
 	}
 
 	// Marshal updated server
+	// keep the id invariant: the json meta id must match the primary key id
```

I think I'd probably prefer throwing an error if someone tries to edit an ID - that seems unintentional and likely indicative of a bug/someone misusing this elsewhere. And I worry this might mask things like messing with meta properties accidentally, which we should be very cautious of for the integrity of data in the registry.

i.e. [fail fast principle](https://martinfowler.com/ieeeSoftware/failFast.pdf)